### PR TITLE
feat: add new auro-menu component

### PR DESCRIPTION
**Pull request by BrentLabasan**
# Alaska Airlines Pull Request

## Summary:

Initial dev work for the new auro-menu component.

## Type of change:

Please delete options that are not relevant.

- [x] New capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/auro-menu.js [hunk 1]
```diff
@@ -35,12 +35,70 @@ class AuroMenu extends LitElement {
...
+          option.removeAttribute('selected');
+        }
+      });
+
+      const elIndex = parseInt(el.getAttribute('index'), 10);
+
+      for (let i = 0; i < options.length; i += 1) {
+        if (elIndex === i) {
+          options[i].setAttribute('selected', '');
+          options[i].querySelector('span').setAttribute('style', 'visibility: visible; margin-right: 8px;');
```
Comment (gusnaughton) [inline src/auro-menu.js:67]: refactor for later - setting a class that contains these attributes would be preferred

File: src/style.scss [hunk 2]
```diff
@@ -10,17 +10,54 @@
...
+  padding: 0;
+  background-color: var(--auro-color-background-lightest);
+
+  font-weight: 600;
+
+  border: 1px solid #F7F7F7 ;
+
+  ::slotted(li) {
+    width: 100%;
+    height: 32px;
```
Comment (gusnaughton) [inline src/style.scss]: use the auro-size for this const

File: src/style.scss [hunk 3]
```diff
@@ -10,17 +10,54 @@
...
+    list-style-type: none;
+    cursor: pointer;
+    padding: 0 8px 0 8px;
```
Comment (gusnaughton) [inline src/style.scss]: auro-size for this const

File: src/auro-menu.js (hunk 1 above)
Comment (BrentLabasan) [reply src/auro-menu.js:67]: Added a TODO comment above saying to refactor.

File: src/style.scss (hunk 2 above)
Comment (BrentLabasan) [reply src/style.scss]: 32px was changed to auro-size variable.

File: src/style.scss (hunk 3 above)
Comment (BrentLabasan) [reply src/style.scss]: 8px was changed to auro-size variable.

**Pull request merged**